### PR TITLE
Use longer timeouts for API checks before trigger a rollback

### LINKED ISSUE #4655: Supervisor's 'Core hasn't started' API timeout check seems to be firing every time
### Describe the issue you are experiencing

#4590 fixed the rollback of HA Core by Supervisor not running correctly... however it seems to now be running too often.

Seemingly, HA Core actually fully starts up within this time -- and the frontend is accessible with sensors working fine. This suggests the API is up and responding. But for some reason, Supervisor fails to detect this and continues to rollback the upgrade. 

Once the rollback is complete, **it detects No API Response again**. But because it's not part of an upgrade, it doesn't roll anything back. And thankfully, it doesn't attempt to restart Core again.

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Steps to reproduce the issue

1. Attempt to upgrade from 2023.11.0b1 to 2023.11.0b2
2. Watch as Supervisor upgrades HA Core, then downgrades it again

### Anything in the Supervisor logs that might be useful for us?

```txt
23-10-30 21:02:32 INFO (MainThread) [supervisor.homeassistant.core] Updating Home Assistant to version 2023.11.0b2
23-10-30 21:02:32 INFO (MainThread) [supervisor.docker.interface] Updating image ghcr.io/home-assistant/generic-x86-64-homeassistant:2023.11.0b1 to ghcr.io/home-assistant/generic-x86-64-homeassistant:2023.11.0b2
23-10-30 21:02:32 INFO (MainThread) [supervisor.docker.interface] Downloading docker image ghcr.io/home-assistant/generic-x86-64-homeassistant with tag 2023.11.0b2.

23-10-30 21:02:39 INFO (MainThread) [supervisor.api.middleware.security] /network/info access from a0d7b954_ssh
23-10-30 21:02:39 INFO (MainThread) [supervisor.api.middleware.security] /network/info access from a0d7b954_ssh
23-10-30 21:02:39 INFO (MainThread) [supervisor.api.middleware.security] /host/info access from a0d7b954_ssh
23-10-30 21:02:39 INFO (MainThread) [supervisor.api.middleware.security] /core/info access from a0d7b954_ssh
23-10-30 21:05:45 INFO (SyncWorker_5) [supervisor.docker.manager] Stopping homeassistant application
23-10-30 21:05:46 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API error: Cannot proxy websocket message of unsupported type: 8
23-10-30 21:05:46 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API connection is closed
23-10-30 21:05:46 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API error: Cannot proxy websocket message of unsupported type: 8
23-10-30 21:05:46 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API connection is closed
23-10-30 21:06:49 INFO (SyncWorker_5) [supervisor.docker.manager] Cleaning homeassistant application
23-10-30 21:06:49 INFO (MainThread) [supervisor.homeassistant.module] Update pulse/client.config: /data/tmp/homeassistant_pulse
23-10-30 21:06:49 INFO (MainThread) [supervisor.docker.homeassistant] Starting Home Assistant ghcr.io/home-assistant/generic-x86-64-homeassistant with version 2023.11.0b2
23-10-30 21:06:49 INFO (MainThread) [supervisor.homeassistant.core] Wait until Home Assistant is ready
23-10-30 21:06:58 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state running
23-10-30 21:06:59 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete

...

23-10-30 21:09:20 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
23-10-30 21:11:39 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'mqtt-scrypted'
23-10-30 21:11:39 INFO (MainThread) [supervisor.auth] Home Assistant not running, checking cache
23-10-30 21:11:41 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'mqtt-appdaemon'
23-10-30 21:11:41 INFO (MainThread) [supervisor.auth] Home Assistant not running, checking cache
23-10-30 21:11:41 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'mqtt-cec'
23-10-30 21:11:41 INFO (MainThread) [supervisor.api.middleware.security] /network/interface/default/info access from dc822f5c_sambanas
23-10-30 21:11:42 INFO (MainThread) [supervisor.api.middleware.security] /network/interface/enp0s18/info access from dc822f5c_sambanas
23-10-30 21:11:42 ERROR (MainThread) [supervisor.services.modules.mqtt] There is already a MQTT service in use from core_mosquitto
23-10-30 21:11:42 INFO (MainThread) [supervisor.auth] Home Assistant not running, checking cache
23-10-30 21:11:52 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'mqtt-glow'
23-10-30 21:11:52 INFO (MainThread) [supervisor.auth] Home Assistant not running, checking cache
23-10-30 21:11:52 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'mqtt-node-red'
23-10-30 21:11:52 INFO (MainThread) [supervisor.auth] Home Assistant not running, checking cache
23-10-30 21:11:52 INFO (MainThread) [supervisor.api.middleware.security] /network/info access from dc822f5c_sambanas
23-10-30 21:11:54 ERROR (MainThread) [supervisor.homeassistant.core] No API response in 5 minutes, assuming core has had a fatal startup error
23-10-30 21:11:54 CRITICAL (MainThread) [supervisor.homeassistant.core] HomeAssistant update failed -> rollback!
23-10-30 21:11:54 INFO (MainThread) [supervisor.homeassistant.core] A backup of the logfile is stored in /config/home-assistant-rollback.log
23-10-30 21:11:54 INFO (MainThread) [supervisor.homeassistant.core] Updating Home Assistant to version 2023.11.0b1
23-10-30 21:11:54 INFO (MainThread) [supervisor.docker.interface] Updating image ghcr.io/home-assistant/generic-x86-64-homeassistant:2023.11.0b2 to ghcr.io/home-assistant/generic-x86-64-homeassistant:2023.11.0b1
23-10-30 21:11:54 INFO (MainThread) [supervisor.docker.interface] Downloading docker image ghcr.io/home-assistant/generic-x86-64-homeassistant with tag 2023.11.0b1.
23-10-30 21:11:55 INFO (SyncWorker_5) [supervisor.docker.manager] Stopping homeassistant application
23-10-30 21:11:55 ERROR (MainThread) [supervisor.homeassistant.api] Error on call http://172.30.32.1:8123/api/core/state: [Errno 104] Connection reset by peer
23-10-30 21:11:55 ERROR (MainThread) [supervisor.homeassistant.api] Error on call http://172.30.32.1:8123/api/core/state: [Errno 104] Connection reset by peer
23-10-30 21:11:59 INFO (SyncWorker_5) [supervisor.docker.manager] Cleaning homeassistant application
23-10-30 21:12:00 INFO (MainThread) [supervisor.homeassistant.module] Update pulse/client.config: /data/tmp/homeassistant_pulse
23-10-30 21:12:00 INFO (MainThread) [supervisor.docker.homeassistant] Starting Home Assistant ghcr.io/home-assistant/generic-x86-64-homeassistant with version 2023.11.0b1
23-10-30 21:12:00 INFO (MainThread) [supervisor.homeassistant.core] Wait until Home Assistant is ready
23-10-30 21:12:10 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state running
23-10-30 21:12:11 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
23-10-30 21:13:25 INFO (MainThread) [supervisor.host.info] Updating local host information
23-10-30 21:13:26 INFO (MainThread) [supervisor.host.services] Updating service information
23-10-30 21:13:26 INFO (MainThread) [supervisor.host.network] Updating local network information
23-10-30 21:13:27 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information
23-10-30 21:13:27 INFO (MainThread) [supervisor.host.manager] Host information reload completed
23-10-30 21:17:03 ERROR (MainThread) [supervisor.homeassistant.core] No API response in 5 minutes, assuming core has had a fatal startup error
```

## System Information

version | core-2023.11.0b1
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.11.6
os_name | Linux
os_version | 6.1.59
arch | x86_64
timezone | Europe/London
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.33.0
Stage | running
Available Repositories | 1400
Downloaded Repositories | 86
HACS Data | ok

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | 5 November 2023 at 00:00
relayer_connected | true
relayer_region | eu-central-1
remote_enabled | true
remote_connected | true
alexa_enabled | true
google_enabled | true
remote_server | eu-central-1-8.ui.nabu.casa
certificate_status | ready
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 11.1.rc1
-- | --
update_channel | beta
supervisor_version | supervisor-2023.10.1
agent_version | 1.6.0
docker_version | 24.0.6
disk_total | 116.7 GB
disk_used | 74.0 GB
healthy | true
supported | true
board | generic-x86-64
supervisor_api | ok
version_api | ok
installed_addons | Mosquitto broker (6.3.1), Ookla Speedtest CLI (1.0.1), Home Assistant Google Drive Backup (0.111.1), Advanced SSH & Web Terminal (15.1.0), MariaDB (2.6.1), Studio Code Server (5.13.0), InfluxDB2 (1.0.2), Caddy 2 (1.5.3), Samba share (10.0.2), Ring-MQTT with Video Streaming (5.6.3), Node-RED (14.6.3), Zigbee2MQTT (1.33.1-1), fluent-bit (1.2.4), Let's Encrypt (4.12.9), JupyterLab (0.12.1), AppDaemon (0.13.6), MQTT Explorer (browser-1.0.1), Network UPS Tools (0.12.2), Tailscale (0.13.1), File editor (5.6.0), diyHue (2.0.10), ESPHome (2023.10.3), RPC Shutdown (2.4), Glances (0.19.4), File editor (5.4.3), scanservjs (v2.26.1-5), Samba NAS (10.0.2-nas4), InfluxDB2 (2.7.0-alpha.3), Zoneminder (1.36.33), Grocy (0.20.1), Music Assistant BETA (2.0.0b74), Scrypted (18-jammy-full.s6-v0.57.0), Zigbee2MQTT Edge (edge)

</details>

<details><summary>Dashboards</summary>

dashboards | 8
-- | --
resources | 40
views | 20
mode | storage

</details>

<details><summary>NextDNS</summary>

can_reach_server | ok
-- | --

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 24 October 2023 at 19:02
-- | --
current_recorder_run | 30 October 2023 at 21:12
estimated_db_size | 957.71 MiB
database_engine | sqlite
database_version | 3.41.2

</details>

<details><summary>Spotify</summary>

api_endpoint_reachable | ok
-- | --

</details>

### Supervisor diagnostics

[config_entry-hassio-6c7ea8e38b0da3aa619bf181cf6d462b.json.txt](https://github.com/home-assistant/supervisor/files/13209975/config_entry-hassio-6c7ea8e38b0da3aa619bf181cf6d462b.json.txt)

### Additional information

<details><summary><h3>HA Core logs</h3></summary>

```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun home-assistant (no readiness notification)
s6-rc: info: service legacy-services successfully started
2023-10-30 21:06:52.817 WARNING (SyncWorker_0) [homeassistant.util.yaml.loader] YAML file /config/themes/m3-xx-yyyy/m3-c11-purple.yaml contains duplicate key "theme-sys-ha-disabled-text-color". Check lines 535 and 537
2023-10-30 21:06:52.817 WARNING (SyncWorker_0) [homeassistant.util.yaml.loader] YAML file /config/themes/m3-xx-yyyy/m3-c11-purple.yaml contains duplicate key "theme-sys-ha-card-background-color-active". Check lines 549 and 550

...

2023-10-30 21:11:44.363 ERROR (MainThread) [frontend.js.latest.202310270] Uncaught error from Safari 17.0 on Mac OS 10.15.7
TypeError: null is not an object (evaluating 'elHaMain.shadowRoot')
/hacsfiles/lovelace-wallpanel/wallpanel.js:2190:10

...

[21:11:55] INFO: Home Assistant Core finish process exit code 256
[21:11:55] INFO: Home Assistant Core finish process received signal 15
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```
</details>

### PR DESCRIPTION
**Pull request by agners**
## Proposed change
Currently we check for Core API access and that the state is running. If this is not fulfilled within 5 minutes, we rollback to the previous version.

It can take quite a while until Home Assistant Core is in state running. In fact, after going through bootstrap, it can theoretically take indefinitely (as in there is no timeout from Core side).

~~So to trigger rollback, rather than check the state to be running, just check if the API is accessible in this case. This prevents spurious rollbacks.~~

However, we can almost guarantee that with all timeouts added up, and some margin, the Core should be up in all but the most obscure setups.

Use a two step method, where we check the API responding first before checking the state. This allows to better understand why the Supervisor chose to rollback, and rolls back the system faster in case there is a serious problem.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #4655
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Comment by bdraco** [comment]
unrelated

```
        # Check if port is up
        if not await self.sys_run_in_executor(
            check_port,
            self.sys_homeassistant.ip_address,
            self.sys_homeassistant.api_port,
        ):
            return False
```

Thats a big unexpected to run in the executor as it could be all non-blocking

File: supervisor/homeassistant/core.py [hunk 1]
```diff
@@ -450,15 +454,25 @@ async def _block_till_run(self, version: AwesomeVersion) -> None:
...
+
                 _LOGGER.info("Detect a running Home Assistant instance")
                 self._error_state = False
                 return
 
         self._error_state = True
         if timeout:
             raise HomeAssistantStartupTimeout(
-                "No API response in 5 minutes, assuming core has had a fatal startup error",
+                "No Home Assistant Core response, assuming a fatal startup error",
```
Comment (bdraco) [inline supervisor/homeassistant/core.py:475]: Maybe change `check_api_state` to `get_api_state` (and make the existing function a wrapper around it) so we can log the state that core was in here which would give us a better hint as to what went wrong if we have a problem here in the future

File: supervisor/homeassistant/api.py [hunk 2]
```diff
@@ -130,22 +130,22 @@ async def get_core_state(self) -> dict[str, Any]:
         """Return Home Assistant core state."""
         return await self._get_json("api/core/state")
 
-    async def check_api_state(self) -> bool:
-        """Return True if Home Assistant up and running."""
+    async def get_api_state(self) -> str | None:
+        """Return True if Home Assistant Core is up and running."""
```
Comment (bdraco) [inline supervisor/homeassistant/api.py:134]:
```suggestion
        """Return state of Home Assistant Core or None."""
```

**Review by bdraco — APPROVED** [review]
LGTM 👍

**Review by mdegat01 — APPROVED** [review]

**Comment by mdegat01** [comment]
Thanks @agners and @bdraco ! 👍

**Pull request merged**